### PR TITLE
Glsurf version 3.3.1

### PR DESCRIPTION
**Pull request by craff**
- 0000014: [General] cleaning of main commands (raffalli) - resolved.
- 0000008: [General] sometime crash on parse error (raffalli) - resolved.
- 0000007: [General] check documentation against parser (raffalli) - resolved.
- 0000012: [General] give mantis address for bug report in opam (raffalli) - resolved.
- 0000004: [General] install as glsurf and not glsurf.opt - resolved.
- 0000010: [General] place parenthesis around & (or use &&) - resolved.
- 0000006: [General] check and update all examples if necessary - resolved.
- 0000005: [General] install the examples and doc in share/doc - resolved.

### CODE REVIEW

**Comment by craff** [comment]
May be the problem with 3.12 is that native camlimages is not compiled/installed ?
 ### stderr ###
 # File "_none_", line 1, characters 0-1:
 # Error: Cannot find file /home/travis/.opam/system/lib/camlimages/camlimages.cmxa
 # make[1]: *** [glsurf.opt] Error 2
 # make: *** [opt] Error 2

**Comment by dsheets** [comment]
Hi, @craff. I think the issue is your build system's specification of the `camlimages` findlib package as a dependency. Older versions of `camlimages` had a bug in their `META` files which indicated that certain archives should be present which aren't built or installed. I believe you only need to build against `camlimages.all_formats`. Alternatively, you could restrict the version of `camlimages` that you depend on.

I played with the program and had some fun. I noticed that in the manual there is a missing semicolon at the end of the `vertices` definition in the cube example which causes it to fail to parse/render.

Finally, if you could remove the `name` and `version` fields from the package description it would be appreciated as these values are implicit in the directory structure.

Thank you for the updated software!

**Comment by craff** [comment]
I am happy to know that you had fun ! I try the dependency on camlimages_all_formats and did the other changed ...

File: packages/glsurf/glsurf.3.3.1/opam [hunk 1]
```diff
@@ -15,7 +13,7 @@ remove: [
   ["rm" "-rf" "%{doc}%/glsurf"]
 ]
 depends: [
-  "camlimages"
+  "camlimages.all_formats"
```
Comment (dsheets) [inline packages/glsurf/glsurf.3.3.1/opam]: Unfortunately, this field is for *opam* dependencies where `camlimages` is the right value. The place that needs to refer to *only* `camlimages.all_formats` and *not* `camlimages` is inside your build system somewhere. Perhaps you have already fixed that as the archive checksum has changed. If so, reverting this line should be all that is necessary.

**Comment by craff** [comment]
Silly me ... thanks, fix is coming

**Comment by craff** [comment]
Well done ! it compiled under 3.12

**Pull request merged**